src/main/java/seedu/address/model/supplier/UniqueSupplierList.java

```diff
@@ -0,0 +1,137 @@
+package seedu.address.model.supplier;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.supplier.exceptions.DuplicateSupplierException;
+import seedu.address.model.supplier.exceptions.SupplierNotFoundException;
+
+/**
+ * A list of suppliers that enforces uniqueness between its elements and does not allow nulls. A supplier is considered
+ * unique by comparing using {@code Supplier#isSameSupplier(Supplier)}. As such, adding and updating of suppliers uses
+ * Supplier#isSameSupplier(Supplier) for equality so as to ensure that the supplier being added or updated is unique in
+ * terms of identity in the UniqueSupplierList. However, the removal of a supplier uses Supplier#equals(Object) so as to
+ * ensure that the supplier with exactly the same fields will be removed.
+ * <p>
+ * Supports a minimal set of list operations.
+ *
+ * @see Supplier#isSameSupplier(Supplier)
+ */
+public class UniqueSupplierList implements Iterable<Supplier> {
+
+    private final ObservableList<Supplier> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Supplier> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent supplier as the given argument.
+     */
+    public boolean contains(Supplier toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameSupplier);
+    }
+
+    /**
+     * Adds a supplier to the list.
+     * The supplier must not already exist in the list.
+     */
+    public void add(Supplier toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateSupplierException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the supplier {@code target} in the list with {@code editedSupplier}.
+     * {@code target} must exist in the list.
+     * The supplier identity of {@code editedSupplier} must not be the same as another existing supplier in the list.
+     */
+    public void setSupplier(Supplier target, Supplier editedSupplier) {
```

Comment:
Similar to comments left on UniqueWarehouseList, might want to change to replaceSupplier